anyway/secrets.py

```diff
@@ -1,26 +1,41 @@
+from typing import Callable
 import os
 import logging
 
 # Never log a secret key!
 # Note that github-secrets are not accessible for PRs from forks.
 
 
-def get(secret_name: str) -> str:
-    env = os.environ.get("FLASK_ENV")
-    logging.info("Reading secret for " + repr(secret_name))
-    secrets_dir = "/run/secrets/"
-    if os.path.isdir(secrets_dir):
-        with open(secrets_dir + secret_name) as secret_file:
-            return secret_file.read()
-    logging.info("No secret dir found.")
-    if env == "development":
-        return os.environ.get(secret_name)
-    else:
-        return os.environ[secret_name]
+def _get_secret_from_dir(secret_name: str) -> str:
+    with open(secrets_dir + secret_name) as secret_file:
+        return secret_file.read()
+
+
+def _get_secret_must_exist(secret_name: str) -> str:
```

Comment:
Not sure what do you mean by `must_exist`. Who's responsibility is to verify it exists?

Reply:
An exception is raised if the secret is not found, by virtue of using the indexing operator. The responsibility of choosing the right function is on the code below, and the responsibility of not trying to use the right function is on the user of the `secrets` module.

The behavior is essentially the same as it used to be in the last year since the introduction of this module. Only now the choice of method happens once, when the module is loaded, and not on every call to `secrets.get()`.

Reply:
Got it :)
I thought that the `_get_secret_must_exist` should be called only when the secret actually exists.